### DISCUSSION TITLE
Creating a quad mesh from a surface

### DISCUSSION THREAD

**Discussion by pempmu** [pyvista/pyvista#2625] [Q&A] [answered]
Hi, I'd like to ask a general question regardin pyvista's functionality. I am interested to know if pyvista can handle mesh generation given a geometry. In my case I would like to create a surface mesh made of Quad elements given a specific contour in a 2D plane:

2D surface given a contour made of lines:
![11111 (1)](https://user-images.githubusercontent.com/63414624/134480862-d1476f80-e2ec-4e72-942e-5f08205b2922.jpg)

2D meshed surface:
![22222 (1)](https://user-images.githubusercontent.com/63414624/134480903-ffb64357-0767-452c-b421-29449694bb70.jpg)

Thank you!

**Comment by akaszynski** [comment, marked as answer]
Pyvista isn't a great meshing tool, but [gmsh](https://gmsh.info/doc/texinfo/gmsh.html) is.  I'd go with that for mesh generation.
